test/upgrade3.test

```diff
@@ -0,0 +1,9 @@
+@ARGS
+--test-repo upgrade.repo
+--test-instdb upgrade3.installed
+--test-world "app a b"
+upgrade a
+@EXPECT
+(1/2) Upgrading b (1 -> 2)
+(2/2) Installing a (2)
```

Comment:
This does not look right. Perhaps the installeddb does not have `a` installed? The test case would make more sense if it results only in upgrading one packages even of upgrades for multiple packages are available.

Reply:
Since I wasn't sure about the expected format I ran `add -u a` to keep the output as a reference. I thought it was thus the expected result. Here `b` is a dependency of `a` so I expected that `b` would also be upgraded (as the last version of `b` is also installed in `basic1.test` when `add a`. Both `a` and `b` code in the repos for this test come from `basic.repo`)

Reply:
The issue was due to `a` which had the same hash than `libold` so I removed `a` in order to only use `b` in the tests.